src/main/java/io/cryostat/agent/WebServer.java

```diff
@@ -82,39 +102,111 @@ class WebServer {
         this.credentials = new Credentials();
         this.callback = callback;
         this.registration = registration;
+        this.fs = fs;
 
         this.agentAuthenticator = new AgentAuthenticator();
         this.requestLoggingFilter = new RequestLoggingFilter();
         this.compressionFilter = new CompressionFilter();
     }
 
-    void start() throws IOException, NoSuchAlgorithmException {
-        if (this.http != null) {
+    void start() {
+        if (this.https != null) {
             stop();
         }
 
-        this.http = HttpServer.create(new InetSocketAddress(host, port), 0);
-        this.http.setExecutor(executor);
-
-        Set<RemoteContext> mergedContexts = new HashSet<>(remoteContexts.get());
-        mergedContexts.add(new PingContext(registration));
-        mergedContexts.stream()
-                .filter(RemoteContext::available)
-                .forEach(
-                        rc -> {
-                            HttpContext ctx = this.http.createContext(rc.path(), wrap(rc::handle));
-                            ctx.setAuthenticator(agentAuthenticator);
-                            ctx.getFilters().add(requestLoggingFilter);
-                            ctx.getFilters().add(compressionFilter);
-                        });
+        try {
+            // initialize new HTTPS server
+            this.https = HttpsServer.create(new InetSocketAddress(host, port), 0);
+            SSLContext sslContext = SSLContext.getInstance("TLS");
```

Comment:
I think this should be `getInstance("TLSv1.2")` or `TLSv1.3`. Same in `MainModule`.

Reply:
https://github.com/cryostatio/cryostat-agent/issues/343